src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -503,10 +506,54 @@ public int callback(zfs_handle_t handle, Pointer arg) {
     }
 
     /**
+     * Grants the specified set of permissions to this dataset.
+     * NOTE that the "zfs_perm_*" libzfs.so routines are not present at
+     * all in OpenZFS, nor even in Sun Solaris 10u8 (last seen in 10u6).
+     * Not sure about Sol10u7 since it was a rather intermediate release.
+     * They may have been replaced with other logic, since zfs_allow_t and
+     * related data types and comments are still present in the headers.
+     */
+    public void allow(ACLBuilder acl) {
```

Comment:
UPDATE: Routines by this name (`zfs_perm_set` and `zfs_perm_remove`) are last present in binaries of Sun Solaris 10u6, and already not present in still-Sun Solaris 10u8 (last Sun release, which is sort of popularly deployed as it still had rather permissive licensing). So newer Solaris and illumos and OpenZFS releases use some other mechanism (maybe same new, maybe different new) to manage the delegation of permissions to manage datasets - one thing for certain, the user-facing feature is still there and it does work (e.g. via CLI tools), so it is a matter of hunting down the needed function names if we need to wrap this in the JAR. Maybe this info is available in 200x history of illumos-gate, as OpenSolaris co-evolved with official Solaris for quite a while (there's probably a commit somewhere there to implement and comment the renaming or refactoring).